File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        l = 0
        r = k-1
        max_list = []

        while r < len(nums):
            max_list.append(max(nums[l:r+1]))
            r += 1
            l += 1

        return max_list

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-0.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
       priority_queue<int> maxHeap;
       for(auto num: nums) {
            maxHeap.push(num);
       } 
        if (k == 1) {
            return maxHeap.top();
        }

        while (true) {
            if (k == 1) {
                return maxHeap.top();
            }
            maxHeap.pop();
            k--;
        }

        return -1;
    }        
};
